### PR TITLE
Added yaml JSON-Schema support

### PR DESCRIPTION
**Pull request by Abrifq**
This may require some testing but looks like it works well against simple schemas defined in YAML.

You can use these files as YAML schema tests: https://github.com/Abrifq/yaml-schema-test

### CODE REVIEW

**Comment by chris48s** [comment]
So this format is.. a JSON schema, but serialized as yaml instead of JSON. Correct? I have not heard of this before, but also.. I guess it should work if it parses to an equivalent JS object.

Just to try and get an idea of the use-case for this..

- Are there any real world examples of projects using JSON schemas serialized as yaml?
- Are there any schemas in this format published on schemastore?

File: src/io.js
Comment (chris48s) [inline src/io.js:12]: Calling `parseFile()` here is not really buying us anything over just invoking `yaml.load()` directly if we have to decide the format first. I might factor out a `parseSchema()` helper with a more restrictive set of formats, but we don't need to do it in this PR.

Comment (Abrifq) [reply src/io.js]: Changed to `yaml.load`.

**Comment by Abrifq** [comment]
Hi, what's blocking this PR for now? 

I know people can be busy and we are not doing this to be paid, I'm aware of that.

If it needs tests, give me a few scenarios and I will try to add them to the specs.

**Comment by chris48s** [comment]
Are you up for finishing this off by adding a test?
https://github.com/chris48s/v8r/pull/364#issuecomment-1737722936

**Comment by Abrifq** [comment]
Yes, just not right now. I'll be travelling for a few days.

**Comment by Abrifq** [comment]
gtg fix more lol

**Comment by Abrifq** [comment]
make sure to review #[6bfa130](https://github.com/chris48s/v8r/pull/364/commits/6bfa130eb82717e35f9f8749ffc52805a598dde8) when you get time

**Comment by chris48s** [comment]
I saw it was updated the other day, but I was a bit unclear if it is still WIP or ready for a review.
I can have a look over the weekend if it is ready..

**Comment by Abrifq** [comment]
It's ready for review.

Make sure to squash/rebase before merging, I had a few rookie mistake commits there 😅

File: src/ajv.js
Comment (chris48s) [inline src/ajv.js:63]: Can you explain the reason for adding this?

I'm not saying it is wrong - I'm just trying to gauge any knock-on effects of adding it.

It is added in a commit that adds tests, but if I remove it, all the tests pass so it seems like it might be an unrelated change.

File: testfiles/schemas/schema.multi-doc.yaml
Comment (chris48s) [inline testfiles/schemas/schema.multi-doc.yaml:10]: Today I learned it is possible to have multiple yaml documents in one file

File: src/ajv.js
Comment (Abrifq) [reply src/ajv.js:63]: Geez, I think it's a leftover. I'll remove that.

**Review by chris48s — APPROVED** [review]

**Comment by chris48s** [comment]
Cool - thanks. I'll put out a 2.1 release with this feature later today

**Pull request merged**
